### ISSUE TITLE
Weirdness with random numbers

### ISSUE THREAD

**Issue by sijnstra** [njm64/zawk#5]
I noticed that the random code doesn't look quite right - I think it is off by 1, as in it can't be 0.
I haven't raised a PR for my hunch because I can't get it to behave correctly anyway. If I run `czech.z3` I get the following as the only failed test:
```
Misc [345]: test...random

ERROR [347]  Expected 148; got 152
```
And if I run `ziptest.z3` I get this kind of output, also clearly incorrect:
```
Randomness check - Type 2-100 or just RETURN to skip this test.
>2
Expected number: 5000

1: 10000
2: 0

Randomness check - Type 2-100 or just RETURN to skip this test.
>2
Expected number: 5000

1: 10000
2: 0

Randomness check - Type 2-100 or just RETURN to skip this test.
>3
Expected number: 3333

1: 5006
2: 4994
3: 0

Randomness check - Type 2-100 or just RETURN to skip this test.
>3
Expected number: 3333

1: 5047
2: 4953
3: 0
```

**Comment by njm64** [comment]
Looks like I forgot to convert the random operand to a signed integer, so seeding wasn't working correctly. I've fixed it in 21d0a30, and czech.z3 passes now. Does that fix it for ziptest.z3?

**Comment by njm64** [comment]
Just tested with ziptest.z3, I see what you mean about the off by one error. This is fixed in 3d1a384.

**Comment by sijnstra** [comment]
Confirmed fixed
~~~
Randomness check - Type 2-100 or just RETURN to skip this test.
>2
Expected number: 5000

1: 4965
2: 5035

Randomness check - Type 2-100 or just RETURN to skip this test.
>3
Expected number: 3333

1: 3293
2: 3407
3: 3300

Randomness check - Type 2-100 or just RETURN to skip this test.
>4
Expected number: 2500

1: 2484
2: 2522
3: 2496
4: 2498

Randomness check - Type 2-100 or just RETURN to skip this test.
>
~~~
